keps/1284-cluster-queue-stop/README.md

```diff
@@ -0,0 +1,133 @@
+
+### Non-Goals
+
+Manage the QuotaReservation and Admission of workloads form the same cohort that might borrow resources from the ClusterQueue in question.
+
+## Proposal
+
+Add a new member in the ClusterQueue implementation `stopPolicy` the presence of which will mark the ClusterQueue as Inactive and it's value will control how the `Admitted` or `Reserving` workloads are affected.
+
+### User Stories
+#### Story 1
+
+As a cluster administrator I want to be able to stop the new admissions in a specific ClusterQueue with the option of Evicting currently admitted Workloads or canceling QuotaReservations.
+
+### Notes/Constraints/Caveats
+Managing the Reservation canceling and Eviction of workloads in other queues from the same cohort that
+are potentially borrowing resources form the stopped queue adds a considerable amount of complexity
+while having a limited added value, therefore these cases are not covered in this first iteration. 
+
+### Risks and Mitigations
+
+## Design Details
+
+### API/ClusterQueue
+
+```go
+type ClusterQueueSpec struct {
+	// ....
+
+	// stopPolicy - if set the ClusterQueue is considered Inactive, no new reservation being
+	// made. 
+	//
+	// Depending on its value, its associated workloads will:
+	//
+	// - None - Workloads are admitted
+	// - HoldAndDrain - Admitted workloads are evicted and Reserving workloads will cancel the reservation.
+	// - Hold - Admitted workloads will run to completion and Reserving workloads will cancel the reservation.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=None;Hold;HoldAndDrain
+	StopPolicy StopPolicy `json:"stopPolicy,omitempty"`
+}
+
+type StopPolicy string
+
+const (
+	None         StopPolicy = "None"
+	HoldAndDrain StopPolicy = "HoldAndDrain"
+	Hold         StopPolicy = "Hold"
+)
+
+
+```
+### Controllers
+#### ClusterQueue
+
+Once the `stopPolicy` is set the cluster queue is marked as inactive with a relevant status message.
+
+#### Workload
+
+If the cluster queue associated to a workload has the `stopPolicy` changed depending on the policy value and state of the
+workload it should Evict or cancel the reservation of the workload.
+
+### Test Plan
+
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+#### Unit Tests
+
+To be added depending on the added code complexity.
+
+#### Integration tests
+
+The `cotrollers/core` suite should check:
+
+1. ClusterQueue - Once the `stopPolicy` is set a ClusterQueue becomes Inactive.
+2. Workload - Once its ClusterQueue `stopPolicy` is set, depending on the value:
+- The Reserving workloads are canceling the reservation.
+- The Admitted workloads get Evicted and the Reserving ones cancel their reservation.
+
+### Graduation Criteria
+
```

Comment:
Can we update the criteria for GA?
Will you graduate this featureGate to GA once the first iteration is completed?

Reply:
I'm not sure it's worth adding a featuregate here, given that default behavior does not change the logic of the queue. But if you think otherwise, let's add